packages/embark-proxy/src/proxy.js

```diff
@@ -19,27 +20,33 @@ export class Proxy {
   }
 
   async serve(endpoint, localHost, localPort, ws, origin) {
-    const start = Date.now();
-    const {host, port} = deconstructUrl(endpoint);
-    await (function waitOnTarget() {
-      return new Promise(resolve => {
-        pingEndpoint(
-          canonicalHost(host),
-          port,
-          ws ? 'ws' : false,
-          'http',
-          origin ? origin.split(',')[0] : undefined,
-          (err) => {
-            if (!err || (Date.now() - start > 10000)) {
-              resolve();
-            } else {
-              timer(250).then(waitOnTarget).then(resolve);
+    if (endpoint !== constants.blockchain.vm) {
+      const start = Date.now();
+      const {host, port} = deconstructUrl(endpoint);
+      await (function waitOnTarget() {
+        return new Promise(resolve => {
+          pingEndpoint(
+            canonicalHost(host),
+            port,
+            ws ? 'ws' : false,
+            'http',
+            origin ? origin.split(',')[0] : undefined,
+            (err) => {
+              if (!err || (Date.now() - start > 10000)) {
+                resolve();
+              } else {
+                timer(250).then(waitOnTarget).then(resolve);
+              }
             }
-          }
-        );
-      });
-    }());
+          );
+        });
+      }());
+    }
 
+    if (endpoint === constants.blockchain.vm) {
+      const ganache = require('ganache-cli');
+      endpoint = ganache.provider(); // Change the endpoint to the Ganache provider
```

Comment:
I don't think this should be here, instead I think starting the VM should be in its own plugin and started in the blockchain module.

I'm not entirely sure what to do at this point - we need the endpoint of the VM for the request manager to get to. Maybe we could hardcode it to `ws://localhost:8546` (the VM default) when we know that we're in the test context and using `'vm'` as the endpoint?